buildrunner/utils.py

```diff
@@ -194,15 +195,22 @@ def write(self, output: Union[bytes, str], color=None):
         # do not colorize output to other streams
         for stream in self.streams:
             try:
-                stream.write(output)
+                if stream.closed:
```

Comment:
There is a bug in ConsoleLogger or in the units tests where I would sometimes see an `I/O operation on closed file.` exception. After digging into it subsequence tests were reusing the same log file in the temporary directory from the previous test which has been closed. This doesn't correct the root cause but makes this operation safer.

Reply:
I was seeing that the `buildrunner` object was being cleaned up but not the `consolelogger` object between tests. This may just be a side effect of how we are testing it and not something that occurs in actual use.